### DISCUSSION TITLE
Release 3.8.0 contains breaking changes (PatternFormatter field name changes)

### DISCUSSION THREAD

**Discussion by kasparthommen** [odygrd/quill#411] [General]
The release [3.8.0](https://github.com/odygrd/quill/releases/tag/v3.8.0) contains breaking changes because the `PatternFormatter` has changed field names. As such, according to [semantic versioning](https://semver.org/), the major version should be bumped not the minor one, i.e., 4.0.0 rather than 3.8.0.

**Comment by odygrd** [comment]
Yes, it's true that according to semantic versioning guidelines, the major version should be bumped when there are breaking changes, like the ones in release 3.8.0 where the PatternFormatter has undergone field name modifications. 

However, throughout the project's lifespan, I usually reserve a major version bump for significant overhauls or extensive redesigns of the library. When there are breaking API changes, I opt for a minor version increment. As for backward-compatible alterations, I stick to patch updates.
